edge-modules/edge-dashboard/src/lib.rs

```diff
@@ -229,6 +234,75 @@ fn restart_module(
     Box::new(response)
 }
 
+fn get_logs(
+    req: HttpRequest,
+    context: web::Data<Arc<Context>>,
+    info: web::Query<AuthRequest>,
+) -> Box<dyn Future<Item = HttpResponse, Error = ActixError>> {
+    let module_id = req.match_info().get("id").unwrap_or("");
+    let api_ver = &info.api_version;
+
+    let response = context
+        .edge_config
+        .as_ref()
+        .map(move |config| {
+            let mgmt_uri = config.connect().management_uri();
+            Either::A(
+                Url::parse(&format!("{}/modules/?api-version={}", mgmt_uri, api_ver))
+                    .map_err(ErrorInternalServerError)
+                    .and_then(|url| ModuleClient::new(&url).map_err(ErrorInternalServerError)) // can't connect to the endpoint
+                    .map(move |mod_client| {
+                        mod_client
+                            .logs(module_id, &LogOptions::new())
```

Comment:
I assume that `client.logs("", ...)` will return error. Shouldn't we prevent calling the client with empty module name earlier, when we take it from request params?